### PR TITLE
feat: allow backups to be encrypted with age

### PR DESCRIPTION
**Pull request by nkcmr**
GPG is known to have usability issues and is generally cumbersome to use. age [0] is a modern alternative to GPG that is designed and implemented by a cryptographer that has worked and continues to work on Golang's crypto packages for years.

Allowing age to be used to encrypt backups dramatically simplifies the backup process.

[0]: https://age-encryption.org/

### CODE REVIEW

File: cmd/backup/encrypt_archive.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"os"
 	"path"
 
+	"filippo.io/age"
```
Comment (m90) [inline cmd/backup/encrypt_archive.go:12]: Question: is it also possible to use the import path `https://github.com/FiloSottile/age` for this package? I do understand the one you use here is the canonical one, but I am slightly worried about specifying a domain here that is under the responsibilty of a single person. Credit cards expire, people go to jail, my build breaks, that kind of stuff. If it's not possible, it's not a blocker.

File: cmd/backup/encrypt_archive.go [hunk 2]
```diff
@@ -17,31 +18,91 @@ import (
 // In case no passphrase is given it returns early, leaving the backup file
 // untouched.
 func (s *script) encryptArchive() error {
-	if s.c.GpgPassphrase == "" {
+	if s.c.GpgPassphrase != "" {
```
Comment (m90) [inline cmd/backup/encrypt_archive.go:21]: Question: what happens when someone provides both GPG and age configuration? Will the file be double encrypted? While there's nothing technically wrong about this, I wonder if it'd be more user friendly to bail on such a setup? Or maybe print a warning.

**Review by m90 — COMMENTED** [review]
Thanks for the contribution, I am happy to add the feature and merge this. I left two questions for my understanding inline. In addition to that it'd be great if you could also add a test case for the new feature. You can probably just copy the existing gpg one and work off of that. If you need to install additional tools for the test container, here's the place to do it: https://github.com/offen/docker-volume-backup/blob/bb11ae035b4d563f7fe86e9bcb7e3dac3c29ed4f/test/Dockerfile#L3-L11

File: cmd/backup/config.go [hunk 3]
```diff
@@ -47,6 +47,8 @@ type Config struct {
 	BackupExcludeRegexp           RegexpDecoder   `split_words:"true"`
 	BackupSkipBackendsFromPrune   []string        `split_words:"true"`
 	GpgPassphrase                 string          `split_words:"true"`
+	AgePassphrase                 string          `split_words:"true"`
+	AgePublicKey                  []string        `split_words:"true"`
```
Comment (m90) [inline cmd/backup/config.go:51]: Since this is a list, should it be named `AgePublicKeys`?

File: cmd/backup/encrypt_archive.go (hunk 2 above)
Comment (nkcmr) [reply cmd/backup/encrypt_archive.go:21]: Good callout, will add a check to ensure configuring age/GPG is `NOT (GPG AND age)`

File: cmd/backup/config.go (hunk 3 above)
Comment (nkcmr) [reply cmd/backup/config.go:51]: Makes sense to me; will change.

Comment (nkcmr) [reply cmd/backup/config.go]: Done: https://github.com/offen/docker-volume-backup/compare/660ddd11aba232fa5ad098e27d7b5ce570bf852d..18b7fb576bf27ba159682ace6953074fc03a70a1

File: cmd/backup/encrypt_archive.go (hunk 2 above)
Comment (nkcmr) [reply cmd/backup/encrypt_archive.go:21]: Done: https://github.com/offen/docker-volume-backup/compare/18b7fb576bf27ba159682ace6953074fc03a70a1..58daebfc43280e7ba9406ae6c5681580c4d3288c

**Comment by m90** [comment]
@nkcmr thanks for the changes. Just to avoid misunderstandings: are you still planning on adding a test for the feature? No intent to pressure you into anything, just want to make sure you're not awaiting some sort of action from my end right now.

**Comment by nkcmr** [comment]
@m90 Yes, sorry for not being more verbose. My morning only had enough time to do those edits before `$WORK`. Will look at the testing setup within the next day or so and get that setup for this change 🙂.

**Comment by m90** [comment]
@nkcmr Sorry for being intrusive, but are you still interested in getting this merged? No worries if your mind wandered elsewhere, I'd just be interested to know if this shall be included in some release in the near future or not. Thanks.

**Comment by m90** [comment]
There's nothing I'd like to release right now, so take your time. Very happy to release this once it's done though, so let me know if I cam help with anything.

File: test/age-publickey/pk-a.txt
[the remark this answers is not in the document]
Comment (nkcmr) [reply test/age-publickey/pk-a.txt:1]: Done!

File: docs/how-tos/encrypt-backups-using-gpg.md
[the remark this answers is not in the document]
Comment (m90) [reply docs/how-tos/encrypt-backups-using-gpg.md:1]: Sounds good, do you still want to add this to the PR? Else I can also do it post-merge.

[the remark this answers is not in the document]
Comment (nkcmr) [reply docs/how-tos/encrypt-backups-using-gpg.md:1]: Done!

**Review by m90 — APPROVED** [review]
This is great, thanks a lot :tophat: . I'll merge and release it once the build is green.

**Comment by m90** [comment]
This is now included in [v2.43.0](https://github.com/offen/docker-volume-backup/releases/tag/v2.43.0)

**Pull request merged**